src/bridge.ts

```diff
@@ -1,24 +1,41 @@
 import { ipcRenderer as IpcRenderer } from 'electron';
 
+import { IPC_CHANNELS } from '@types';
+
 type Unsubscribe = () => void;
 type Listener = (...args: any[]) => void;
 
 export interface IIpcBridge {
   send(channel: string, data: any): void;
   subscribe(channel: string, listener: Listener): Unsubscribe;
+  invoke(channel: string, ...args: any[]): Promise<any>;
 }
 
+const isValidChannel = (channel: string) => Object.values(IPC_CHANNELS).includes(channel);
+
 export const IpcBridge = (ipcRenderer: typeof IpcRenderer): IIpcBridge => ({
   send: (channel: string, data: any) => {
+    if (!isValidChannel(channel)) {
+      throw Error('Invalid Channel');
```

Comment:
Add `Logger`

Reply:
Huh? Don't we want to throw here? We don't want execution to continue.

Reply:
Add comment: why we fail hard